nexusutils/dataconverter/helpers.py

```diff
@@ -406,6 +409,9 @@ def validate_data_dict(template, data, nxdl_root: ET.Element):
                 index_of_at = nxdl_path.rindex("@")
                 nxdl_path = nxdl_path[0:index_of_at] + nxdl_path[index_of_at + 1:]
 
+            if nxdl_path in nxdl_path_set:
```

Comment:
This will skip over template entries with the same NXDL definition and will not validate the second occurrence. Are we sure we need this?

Reply:
@shabihsherjeel, In XPS if it takes around 2 hours to validate each NXDL definition. That is the reason to skip the validation step for the same nxdl occurrence. And I assumed that once the nxdl path is correct it will be correct for all nxdl paths.